examples/step-57/doc/intro.dox

```diff
@@ -0,0 +1,319 @@
+        F \\
+        0 \\
+      \end{array}
+    \right)
+@f}
+with a parameter $\gamma$ and an invertible matrix W.
+
+Denoting the system matrix of the new system by $G$ and the right-hand
+side by $b$, we solve it iteratively with right preconditioning
+$P^{-1}$ as $GP^{-1}y = b$, where
+
+@f{eqnarray*}
+P^{-1} = \left(\begin{array}{cc} \tilde{A} & B^T \\ 
+                                 0 & \tilde{S} \end{array}\right)^{-1},
+@f}
+
+with $\tilde{A} = A + \gamma B^TW^{-1}B$ and $\tilde{S}$ is the
+corresponding Schur complement $\tilde{S} = B^T \tilde{A}^{-1} B$. We
+let $W = M_p$ where $M_p$ is the pressure mass matrix, then
+$\tilde{S}^{-1}$ can be approximated by
+
+@f{eqnarray*}
+\tilde{S}^{-1} \approx -(\nu+\gamma)M_p^{-1}.
+@f}
+See [1] for details.
+
+We decompose $P^{-1}$ as
+
+@f{eqnarray*} 
+P^{-1} =
+\left(\begin{array}{cc} \tilde{A}^{-1} & 0 \\ 0 & I \end{array}\right)
+\left(\begin{array}{cc} I & -B^T \\ 0 & I \end{array}\right)
+\left(\begin{array}{cc} I & 0 \\ 0 & \tilde{S}^{-1} \end{array}\right).
+@f}
+
+Here two inexact solvers will be needed for $\tilde{A}^{-1}$ and
+$\tilde{S}^{-1}$, respectively(see [1]). Since the pressure mass
+matrix is symmetric and positive definite,
+CG with ILU as a preconditioner is appropriate to use for $\tilde{S}^{-1}$. For simplicity, we use 
+the direct solver UMFPACK for $\tilde{A}^{-1}$. The last ingredient is a sparse
+matrix-vector product with $B^T$. Instead of computing the matrix product
+in the augmented Lagrangian term in $\tilde{A}$, we assemble Grad-Div stabilization
+$(\nabla \cdot \phi _{j}, \nabla \cdot \phi _{i}) \approx (B^T
+M_p^{-1}B)_{ij}$, as explained in [2].
+
+<h3> Test Case </h3>
+
+Here we use the Lid driven cavity flow as our test case, see [3] for details. 
+The computational domain is the unit square and the right-hand side
+$f=0$. The boundary condition is
+
+@f{eqnarray*} 
+(u(x, y), v(x,y)) &=& (1,0)   
+  \qquad\qquad \textrm{if}\ y = 1 \\
+  (u(x, y), v(x,y)) &=& (0,0)
+  \qquad\qquad \textrm{else}.
+@f}
+
+When solving this problem, the error consists of the nonlinear error (from
+Newton's iteration) and the discrete error (depending on mesh size). The
+nonlinear part decreases with each Newton iteration and the discrete error
+reduces with mesh refinement. In this example, the solution from the coarse
+mesh is transferred to successively finer meshes and used as an initial
+guess. Therefore, the nonlinear error is always brought below the tolerance of
+Newton's iteration and the discrete error is reduced with each mesh
+refinement.
+
+Inside the loop, we involve three solvers: one for $\tilde{A}^{-1}$,
+one for $M_p^{-1}$ and one for $Gx=b$. The first two
+solvers are invoked in the preconditioner and the outer solver gives us
+the update term. Overall convergence is controlled by the nonlinear residual 
+and Newton's method does not require an exact Jacobian, so for the outer 
+linear solver we employ FGMRES with a relative tolerance of only 1e-4. 
```

Comment:
That's not what one typically means when one says that one uses an inexact Jacobian. What you describe is called a truncated or inexact Newton solve.